src/net/udp.rs

```diff
@@ -253,12 +264,17 @@ impl UdpSocket {
         self.inner.read_fixed(buf).await
     }
 
-    /// Write some data to the socket from the buffer, returning the original buffer and
-    /// quantity of data written.
+    /// Writes data into the socket from the specified buffer.
+    ///
+    /// Takes ownership of the buffer for the duration of the Future.
```

Comment:
It's true that it takes ownership of the buffer for the duration of the future. But are we going to say that for each of our ops that uses a supplied buffer? Maybe a separate commit should address a nice comment like this for each of the output methods of the library?

Reply:
I'm of two minds about this:  

1. To an experienced rust dev that is paying attention, they'd see from the signature that the  buffer is passed by value, or their rust-analyzer would certainly let them know.  So we shouldn't bother to point it out. 

-or-

2.1.   I'm an experience Rust dev, but I rarely pay that close of attention to signatures in docs.
2.2.  This is a significant departure from pretty much every other io call in the Rust ecosystem. 
2.3.  It has architectural implications.  A person has to change the design of their program (sometimes significantly) to work with buffers that are passed by value. 

For these reasons, I'm definitely leaning towards a rule: 

"We should make any IO that passes a buffer by something other than Reference explicitly state such in the docs -  UNTIL - the new set of IO traits that Nick Cameron et al. is working on has been completed and adopted."

Reply:
Fair that you see two sides to this. But I don't want to change all the docs at this time because of the complication that you see. It would mean changes to many places. Such a change would warrant its own issue and possible PR and not be mixed in with comment changes were clearly poorer english is being replaced with better english. This repo has lots of bugs and features that need addressing, and will even go through several more big rewrites I suspect, and I don't think we should be spending so much time trying to make the comments of some functions so overly descriptive. I prefer good long comments for people who are trying to make development changes to the code base at this point.

Reply:
Ok.